pkgs/development/compilers/llvm/15/default.nix

```diff
@@ -15,24 +15,73 @@
     then null
     else pkgs.bintools
 , darwin
+# LLVM release information; specify one of these but not both:
+, gitRelease ? null
+  # i.e.:
+  # {
+  #   version = /* i.e. "15.0.0" */;
+  #   rev = /* commit SHA */;
+  #   rev-version = /* human readable version; i.e. "unstable-2022-26-07" */;
+  #   sha256 = /* checksum for this release, can omit if specifying your own `monorepoSrc` */;
+  # }
+, officialRelease ? { version = "15.0.2"; sha256 = "sha256-eGvRLlW8mH2KxVvEzauuO+askfB+WqNsqL5x5rPT1Es="; }
+  # i.e.:
+  # {
+  #   version = /* i.e. "15.0.0" */;
+  #   candidate = /* optional; if specified, should be: "rcN" */
+  #   sha256 = /* checksum for this release, can omit if specifying your own `monorepoSrc` */;
+  # }
+# By default, we'll try to fetch a release from `github:llvm/llvm-project`
+# corresponding to the `gitRelease` or `officialRelease` specified.
+#
+# You can provide your own LLVM source by specifying this arg but then it's up
+# to you to make sure that the LLVM repo given matches the release configuration
+# specified.
+, monorepoSrc ? null
```

Comment:
IMO it'd be best to do this in another PR, and then you can add it to all the `llvmPackages_*` too :)

Reply:
I'm going to leave the defensive check part of the commit in since that actually would trigger rebuilds if added in later.

I think it's a decent safeguard to have anyways.

https://github.com/NixOS/nixpkgs/blob/5a831a41759079fcc0e8bb43e3bcc6f82c59c0a2/pkgs/development/compilers/llvm/15/llvm/default.nix#L114-L144

Reply:
> because of the two nested package sets

I don't understand why we have this and I haven't worked out the right way to override part of the package and have it picked up by the others. It's very weird.

The other package sets we have that have multiple layers seem to be constructed differently with makeScope for both parts and llvm only has it for the inner tools one :confused:

Reply:
@LunNova can you open an issue for this? I can add it to the list of follow-up work in #213033.

It's very possible I'm missing something but I can't think of any reasons not to – at the very least – add an outer `makeExtensible` so that it's a little easier to override packages in the set without recreating the logic connecting the tools and library package sets.

Reply:
Split out the monorepo override stuff into it's own commit: https://github.com/NixOS/nixpkgs/pull/194634/commits/d231d18e4aa5e1d00f86b4f484f9e4344538e3ea